daemon/mgr/container.go

```diff
@@ -1166,6 +1166,33 @@ func (mgr *ContainerManager) Remove(ctx context.Context, name string, options *t
 		if err := mount.Unmount(c.BaseFS, 0); err != nil {
```

Comment:
Actually I do not think make the code block in the condition of `if c.RootFSProvided` is a good idea. It is a very particular case. I would like to see this part of code to be encapsulated into a single function, like `func DealRootfsProvided(a,b,c string)`..... Which will definitely make code more readable. @HusterWan 

WDYT?

Reply:
Make sense